contracts/vaults/BaseVaults/RibbonThetaVault.sol

```diff
@@ -322,6 +335,19 @@ contract RibbonThetaVault is RibbonVault, RibbonThetaVaultStorage {
         );
     }
 
+    /**
+     * @notice Stakes a users vault shares
+     * @param numShares is the number of shares to stake
+     */
+    function stake(uint256 numShares) external nonReentrant {
```

Comment:
Imagine Alice has 50 shares of rETH-THETA. She does a redeemMax() for 50 shares. She then calls `stake(50 shares)`. 

This will revert because we are trying to redeem shares that have already been redeemed, but Alice staking her separately redeemed shares should be a valid flow.  

I recommend replacing `_redeem(numShares, false);` with 

```
(uint256 heldByAccount, uint256 heldByVault) = shareBalances(msg.sender); 
require(numShares <= heldByAccount.add(heldByVault), "numShares greater than shares owned!");
if(heldByAccount < numShares){
      _redeem(numShares.sub(heldByAccount), false);
}
```

Reply:
good idea, just added it here (along with tests for it): https://github.com/ribbon-finance/ribbon-v2/pull/194/commits/92bfe9b03ee5c687cb20a46e179a713040d6aee6
I didn't add the require line since it would anyways revert when calling `_transfer` if numShares were to exceed the shares the user owned